server_test.go: TestLocalFilePluginRegister: Don't start server

evanj opened the pull request:
#### Summary

This test only needs to create the server from the configuration, it
does not need to start it. Previously, this test did not shut down
the server, which caused lots of confusing log output for all tests
that ran after it, since the server was running continuously.


#### Motivation

I can't currently run the unit tests on my Mac. This is not a critical problem, but it does make the logs much cleaner.

#### Test plan

This is a test-only change.

cory-stripe commented:
👍 

Oh man, you're on quite a spree! Again, sorry for all the rebasing needed here. Lemme know and I will handle all that on our end if you prefer.

evanj commented:
The spree was due to the comment on #133, causing me to revisit this. It turns out a bunch of tests didn't work for me, so I figured out some way to make them work. It turns out I haven't had to touch our Veneur configuration for a couple months, since I figured out the memory leak in the TCP code I added.

Rebased and passed on TravisCI!

cory-stripe commented:
I'm trying to reproduce's @aditya-stripe's success at rebasing other people's branches. Do you have that box checked that lets me write to your branch? If not, please do so for this and your other two? Every time I merge I'm screwing up your PR and can't merge 'em :(

evanj commented:
Huh, yeah, it says "Allow edits from maintainers". No worries: I'm happy to keep rebasing, although we will have some slow turnaround since I don't check my personal Github that often during the day :)

cory-stripe commented:
I figured it out. My fault. Will be going on a merging spree now!
